pkg/workspace/workspace.go

```diff
@@ -283,16 +283,15 @@ func (ws *Workspace) HasCatalog(name string) (bool, wfapi.Error) {
 	return true, nil
 }
 
-// Create a new catalog.
-// This only creates the catalog and does not open it.
+// CreateCatalog creates a new catalog.
+// CreateCatalog only creates the catalog and does not open it.
 //
 // Errors:
 //
-//     - warpforge-error-io -- when reading or writing the catalog directory fails
-//     - warpforge-error-catalog-invalid -- when the catalog already exists
-//     - warpforge-error-catalog-name -- when the catalog name is invalid
+//    - warpforge-error-io -- when reading or writing the catalog directory fails
+//    - warpforge-error-file-exists -- when the catalog already exists
```

Comment:
I'd probably rather we have "warpforge-error-already-exists" and leave the file detail out of the headline?

Reply:
Should it still record path or any other details?

Reply:
```
		Message:    "already exists"
```
Seems a bit sparse.

Reply:
I think putting a path or mention files in the message text or details map is fine.  More info is better in those!  I'm only objecting to mentioning files in the code itself.

Reply:
So the constructor implicitly means files but the error code itself doesn't. I.E. The code is reusable for multiple constructors?

Reply:
We could then change `warpforge-error-catalog-item-already-exists` to return a code of `warpforge-error-already-exists`. I'm not really sure if that's _favorable_ but would reduce raw quantities of error codes.

Reply:
Also `warpforge-error-already-exists` seems excessive compared to `warpforge-error-exists`.

Reply:
- the constructors could definitely be re-steered.  I think it's fine to have more than one constructor that produces an error with the same code, but with different parameters because it's describing a slightly different origin story for the error.
- It's hard to say for sure, because we're still figuring out what good heuristics for error codes are, but I think I'm :+1: on `warpforge-error-catalog-item-already-exists` -> `warpforge-error-already-exists`, yes!  (In both cases, the remedy action would not be to retry, but would involve changing a parameter in some direction to avoid a collision... so, for these things to have the same error code seems logically valid.  And also because they tend not to emerge up along the same code paths, it is unlikely to be confusing.  It's passing every heuristic I can think of at the moment!)
- :-1: on stripping the word "already" out.  Maybe this is personal taste, but I think that _does_ provide useful information to a human being.  Sure, if words had no inherent meaning and we gziped everything, `already-exists` and `exists` could be collapsed to the same shortcode in practice.  But that's not how humans process things.